src/components/utils/useSelect/useOpenState.ts

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import type {UseOpenProps} from './types';
+
+export const useOpenState = (props: UseOpenProps) => {
+    const [open, setOpenState] = React.useState(props.defaultOpen || false);
+    const isOpenControlled = React.useMemo(() => typeof props.open === 'boolean', [props.open]);
+    const {onOpenChange, onClose} = props;
+
+    React.useEffect(() => {
+        if (!isOpenControlled) return;
+
+        setOpenState(props.open as boolean);
+    }, [props.open, isOpenControlled]);
+
+    const setOpen = React.useCallback(() => {
+        if (isOpenControlled || open) return;
+        setOpenState(true);
+        onOpenChange?.(true);
+    }, [setOpenState, isOpenControlled, onOpenChange, open]);
+
+    const setClose = React.useCallback(() => {
+        if (!open) return;
+        onClose?.();
+        if (isOpenControlled) return;
+        setOpenState(false);
+        onOpenChange?.(false);
+    }, [setOpenState, onClose, onOpenChange, isOpenControlled, open]);
+
+    const toggleOpen = React.useCallback(() => {
+        if (open) {
+            setClose();
+        } else {
+            setOpen();
+        }
+    }, [setClose, open, setOpen]);
+
+    return {open, setOpen, setClose, toggleOpen};
```

Comment:
`setOpen` changes looks a little bit major. Can we do it without changing current api? Maybe add new methods with different names and deprecate old setOpen

Reply:
Right now there is only one usage place (inside useSelect hook).
I didn't change the items of useSelect return object just extend it.

old items: `open, setOpen`
new items: `setClose, toggleOpen`